bodhi/server/services/updates.py

```diff
@@ -583,3 +592,37 @@ def waive_test_results(request):
         request.errors.add('body', 'request', str(e))
 
     return dict(update=update)
+
+
+@update_get_test_results.get(schema=bodhi.server.schemas.GetTestResultsSchema,
+                             validators=(validate_update_id,
+                                         validate_acls),
+                             permission='read', renderer='json',
+                             error_handler=bodhi.server.services.errors.json_handler)
+def get_test_results(request):
+    """
+    Get the test results on a given update when gating is on.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: A dictionary mapping the key "update" to the update.
+    """
+    update = request.validated['update']
+
+    decision = None
+    try:
+        decision = update.get_test_gating_info()
```

Comment:
It would be good to also explicitly catch ```requests.exceptions.Timeout``` and set a 504 when that happens.

Reply:
The reason I caught the high level exception is that if we set a 504 when querying greenwave, we won't be able to distinguish between a 504 coming from bodhi vs a 504 coming from greenwave.
So I went with: if the error comes from greenwave, always return a 502 and include in the JSON returned the exact error (500, 504, 502...).

The test at: https://github.com/fedora-infra/bodhi/pull/2392/files#diff-1ff79f6cd36ea2ba9a09899a2799ae1dR5154 shows this behavior.

I can definitely adjust if you prefer though